Automattic/wp-calypso
# Site Migration: Don't retry to install migration plugin if it's already installed

closes to  #90978

## Proposed Changes

* Fix an issue on wpcom HTTP client when it is running inside nodejs (test env context). The original version was not enable us to capture errors when we ran the code on the test suite + http_envelop=1.
* Update plugin installation process to assume as a success if the endpoint returns the error `PluginAlreadyInstalledError`

## Testing Instructions
* Checkout this branch in your local env
* Start a migration `/setup/migration-signup` 
* Open the developer tools > network
* Go to the instructions page
* Fix any jetpack connection issue to enable the plugin installation if need
* Wait until the plugin installation is completed. 
* Go to your site and disable the `migrate-guru` plugin
* Apply this [patch] (https://github.com/user-attachments/files/16381662/force_install.patch) in your local env to force the code to try to install the plugin again. `git apply force_install.path`. 
* Reload the instructions page and check if there is only one call to `/sites/${ siteId }/plugins/${ pluginSlug }/install`.
* The endpoint will return the error `PluginAlreadyInstalledError` 
* The UI should not retry the request and assume it was installed correctly.
* The UI should activate the plugin after trying to install it.  

> [!NOTE]  
> The plugin activation endpoint is idempotent. The API doesn't return errors if we try to active the same plugin multiple times.


## Pre-merge Checklist
